prometheus/testutil/promlint/promlint.go

```diff
@@ -29,7 +29,8 @@ import (
 // A Linter is a Prometheus metrics linter.  It identifies issues with metric
 // names, types, and metadata, and reports them to the caller.
 type Linter struct {
-	r io.Reader
+	r   io.Reader
```

Comment:
I would comment this is `oneof` between two. I was bit confused and I can imagine package devs might be as well..

Not relevant to package users

Reply:
It would actually work with both having values, just that you don't have a constructor doing so (but we might have in the future). I'll add a comment explaining that.